src/main/java/com/home/eshop/dao/DaoCache.java

```diff
@@ -0,0 +1,83 @@
+package com.home.eshop.dao;
+
+import com.home.eshop.model.Bike;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class DaoCache implements DaoInterface {
+    private Map<Integer, Bike> mapBikes;
+    private BikesDao bikesDao;
+
+    public DaoCache(String path) {
+        addBikesMap(path);
+    }
+
+    private void addBikesMap(String path) {
+        if (bikesDao == null) {
+            bikesDao = new BikesDao(path);
+            mapBikes = bikesDao
+                    .findAll()
+                    .stream()
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toMap(Bike::getId, Function.identity()));
+        }
+    }
+
+    private static void bikeNotFound() {
+        System.out.println("Bike not found");
+    }
+
+    @Override
+    public int change(Bike bike) {
+        Optional<Bike> optionalBike = Optional.ofNullable(mapBikes.get(bike.getId()));
+        optionalBike.ifPresent(bikeChang -> {
+            mapBikes.remove(bikeChang.getId());
+            mapBikes.put(bikeChang.getId(), bikeChang);
+        });
+        optionalBike.orElse(bikesDao.findOne(bike.getId()));
```

Comment:
what the logic behind it? explain me in the chat

Reply:
public int update(Bike bike) {
 1   Optional<Bike> optionalBike = Optional.ofNullable(mapBikes.get(bike.getId()));
 2   optionalBike.ifPresent(bikeChang -> {mapBikes.put(bikeChang.getId(), bikeChang);});
 3   optionalBike.orElse(bikesDao.findOne(bike.getId()));
 4   optionalBike.ifPresent(bikeChang -> bikesDao.update(bikeChang));
 5   return bike.getId();}

such logic
1. get (Object or Null) is map ;
2. if  (Object) put the Map ; 
3. if         (Null) get (Object or Null) is txt ;
4. if  (Object) put the txt ;
5. return key ;

old version
 4    optionalBike.ifPresentOrElse(bikeChang -> bikesDao.update(bikeChang), BikesCache::bikeNotFound);
4. if  (Object) put the txt ,if(null) print ("Bike not found")

Reply:
Mission Complete

Reply:
Corrected the mistake.

2 optionalBike.ifPresent(bikeChang -> {mapBikes.put(bikeChang.getId(), bike);});
3 optionalBike.orElse(bikesDao.findOne(bike.getId()));
4 optionalBike.ifPresent(bikeChang -> bikesDao.update(bike));

Reply:
Mission Complete